### ISSUE TITLE
add Gaia-style columns to Galaxia/MWS mocks?

### ISSUE THREAD

**Issue by moustakas** [desihub/desitarget#352]
@apcooper I don't know how much effort would be involved, or if this activity is already in progress, but are there any plans to add Gaia-style columns to the Galaxia mocks?  As of #338, our standard star selection now *requires* Gaia, so this is a necessary update/upgrade to make our mocks and simulated datasets as accurate as possible.

Thanks--

**Comment by moustakas** [comment]
Apologies for the premature post, but it looks like there are already some Gaia catalogs in `/global/projecta/projectdirs/desi/mocks/mws/galaxia/alpha/v0.0.5` with columns:
```
['V_LUPTON_NODUST',
 'I_LUPTON_NODUST',
 'V_LUPTON',
 'I_LUPTON',
 'G_GAIA',
 'G_GAIA_RVS',
 'PM_RA_GAIA_ERROR',
 'PM_DEC_GAIA_ERROR',
 'PM_RA_STAR_GAIA',
 'PM_DEC_GAIA',
 'PARALLAX_GAIA_ERROR',
 'PARALLAX_GAIA',
 'V_HELIO_GAIA_ERROR_G0V',
 'V_HELIO_GAIA_G0V',
 'V_HELIO_GAIA_ERROR_F0V',
 'V_HELIO_GAIA_F0V',
 'V_HELIO_GAIA_ERROR_K1III',
 'V_HELIO_GAIA_K1III',
 'V_HELIO_GAIA_ERROR_K1IIIMP',
 'V_HELIO_GAIA_K1IIIMP']
```
However, we would need all the Gaia columns listed in the sweep files--
  http://www.legacysurvey.org/dr7/files/#sweep-catalogs 
in order to select standards --
https://desi.lbl.gov/trac/wiki/TargetSelectionWG/TargetSelection#SpectrophotometricStandardStarsSTD

Looking a bit more closely, I can ignore / mock up the `GAIA_ASTROMETRIC_EXCESS_NOISE` and `GAIA_DUPLICATED_SOURCE` columns, so all I need is `GAIA_PHOT_BP_MEAN_MAG`, `GAIA_PHOT_BP_MEAN_FLUX_OVER_ERROR`, `GAIA_PHOT_RP_MEAN_MAG`, and `GAIA_PHOT_RP_MEAN_FLUX_OVER_ERROR`, assuming the G-band magnitudes, proper motions, and parallaxes in the Galaxia mocks are in the same units as the sweep files.

**Comment by apcooper** [comment]
@moustakas: OK, clearly we're at the stage now where the sweep data model is pretty stable and it's worth making the mocks look more like the real catalogs. I'll make it a priority to get that done asap. There are a couple of other small updates to the mocks pending from other issues that I hope I can fix at the same time. 

As for these: `GAIA_PHOT_BP_MEAN_MAG, GAIA_PHOT_BP_MEAN_FLUX_OVER_ERROR, GAIA_PHOT_RP_MEAN_MAG, and GAIA_PHOT_RP_MEAN_FLUX_OVER_ERROR`: it looks like those columns are only used for selecting the WDs (the only exception I can see is the requirement that BP-RP be `not nan` for standards). 

My understanding from Boris G. is that he's not planning on using Gaia colour to select WDs now -- or at least not in any simple way. I'll ask him what he thinks is the best way to mock up the WD selection now. I'll cc you on that.

**Comment by moustakas** [comment]
Note that this issue is a duplicate of #326, where I asked the exact same question.

("I grow old...I grow old...I shall wear the bottoms of my trousers rolled.")

**Comment by moustakas** [comment]
@apcooper  Will the "nearby" (`100pc/v0.0.3/mock_100pc.fits`) and white dwarf (`wd/v0.0.2/mock_wd.fits`) mocks also be updated to include Gaia columns?  That would be very helpful.

**Comment by apcooper** [comment]
> As a minimum immediate fix could you make a version of the mocks which have the Gaia catalogs (i.e., gaia_mock_allsky_galaxia_desi-8-???.fits) merged with the older Galaxia columns (mock_allsky_galaxia_desi-8-???.fits) so the code only has to deal with one catalog? 

I've done this minimal fix, renaming a bunch of the columns and converting error to invvar for PMRA, PMDEC and PARALLAX (i.e. 1/var = 1/s^2 where s is the error reported in the previous version of the files). I haven't created a new README yet so I hope it's obvious what is what. PMRA *includes* the cos(dec) term, I'm not sure if this is the case for the sweeps. All the columns that are not direct observables have 'TRUE_' prepended.

This preliminary version will be in the usual place under 0.0.6 by tomorrow (will post here when they're ready). There is no version of this for the superfaint catalogs yet (are those still being used, and if so how? They are something of a pain to keep in sync...). 

I'm away for the next week but will continue to tackle this as soon as I get back. Would any of the following be useful:
* Moving superfluous 'TRUE_xxx' columns to separate files that `desitarget` can ignore?
* Doing the SDSS -> DECALS conversion when making the mocks and reporting fluxes as Tractor?

**Comment by apcooper** [comment]
Those files are at NERSC now.

**Comment by moustakas** [comment]
> This preliminary version will be in the usual place under 0.0.6 by tomorrow (will post here when they're ready). There is no version of this for the superfaint catalogs yet (are those still being used, and if so how? They are something of a pain to keep in sync...).

The superfaint catalogs are being used to simulate stellar contaminants for the other target classes, so they're pretty important, especially for the QSOs. I wonder if I can just bootstrap from the existing MWS_MAIN catalogs somehow...

> * Moving superfluous 'TRUE_xxx' columns to separate files that desitarget can ignore?
> * Doing the SDSS -> DECALS conversion when making the mocks and reporting fluxes as Tractor?

No, I handle this all in code.  The output of `select_mock_targets` is a catalog that looks just like an output targeting catalog and the different bandpasses are handled when generating spectra -- each spectrum is normalized in the SDSS bandpass and then the DECaLS/MzLS+BASS fluxes and colors are synthesized.

**Comment by moustakas** [comment]
> I've done this minimal fix, renaming a bunch of the columns and converting error to invvar for PMRA, PMDEC and PARALLAX (i.e. 1/var = 1/s^2 where s is the error reported in the previous version of the files). I haven't created a new README yet so I hope it's obvious what is what. PMRA includes the cos(dec) term, I'm not sure if this is the case for the sweeps. All the columns that are not direct observables have 'TRUE_' prepended.

It would be very helpful @apcooper if the units of these various columns could be explicitly stored in the FITS files.  Thanks.

**Comment by geordie666** [comment]
`PMRA` (and all Gaia columns) in the sweeps are as in the Gaia data model:

https://gea.esac.esa.int/archive/documentation/GDR2/Gaia_archive/chap_datamodel/sec_dm_main_tables/ssec_dm_gaia_source.html

`Proper motion in right ascension μα*≡μαcosδ of the source in ICRS at the reference epoch ref_epoch. This is the local tangent plane projection of the proper motion vector in the direction of increasing right ascension.`

**Comment by apcooper** [comment]
Hi @moustakas,

A 'full' version of the v0.0.6 galaxia mock has replaced the temporary version at NERSC.  The temporary version is in `v0.0.6_prelim`. Apologies for the delay in getting this done.

This new version:
- has a footprint derived from the (soon-to-be superseded?) standard tiles file, but *with a 1-healpixel dilation* as a hedge against edge effects and updates to the tiles;
- has column names attempting to match the tractor names for the observables, as in the temporary version;
- has units associated with almost all the columns. A few (e.g. FeH, logg, Teff) don't have units because astropy has some issue with saving `dex` units to FITS. I just give those 'dimensionless' units instead. Let me know if you want me to work round that some other way;
- fixes a bug that might have affected the selection of standards (hopefully not much): the values in the Gaia G mag column were actually V mag.

The `FCOVER` fractions in the header is computed from the current tiles, so would need to be updated if the tiles change.

Let me know if these are OK, or if there are any other changes you need urgently. If everything is alright I'll proceed with the 'superfaint' equivalent (it shouldn't take very long).

**Comment by moustakas** [comment]
@apcooper Thanks for these changes.  Let me look at these files after #395 has been merged, together with some other pending updates to the mocks.

As for the superfaint mocks, it would be great to have a consistent data model with respect to the spectral/stellar properties, but I don't think the Gaia data are necessary since we only select standard stars from the `MWS_MAIN/Galaxia` mocks (owing to the faint cut).

**Comment by moustakas** [comment]
> has units associated with almost all the columns. A few (e.g. FeH, logg, Teff) don't have units because astropy has some issue with saving dex units to FITS. I just give those 'dimensionless' units instead. Let me know if you want me to work round that some other way;

Note that the astropy-compatible unit is `Dex`, not `dex`.

**Comment by moustakas** [comment]
@apcooper The `v0.0.6_superfaint` catalogs seem to be missing the radial velocity (`TRUE_RADIAL_VELOCITY` in the `MWS_MAIN` mocks). Or am I missing something obvious?  Thanks.
```
  file: /Users/ioannis/work/desi/mocks/mws/galaxia/alpha/v0.0.6_superfaint/healpix/8/4/406/mock_superfaint_allsky_galaxiasi_b10-8-406.fits
  extension: 1
  type: BINARY_TBL
  rows: 315854
  column info:
    X                   f4  
    Y                   f4  
    Z                   f4  
    L                   f4  
    B                   f4  
    RA                  f8  
    DEC                 f8  
    TRUE_DHELIO         f4  
    TRUE_DM             f4  
    ABV                 f4  
    TRUE_MAG_U_SDSS_NODUST
                        f4  
    TRUE_MAG_U_SDSS     f4  
    MAG_U_SDSS          f4  
    TRUE_MAG_G_SDSS_NODUST
                        f4  
    TRUE_MAG_G_SDSS     f4  
    MAG_G_SDSS          f4  
    TRUE_MAG_R_SDSS_NODUST
                        f4  
    TRUE_MAG_R_SDSS     f4  
    MAG_R_SDSS          f4  
    TRUE_MAG_I_SDSS_NODUST
                        f4  
    TRUE_MAG_I_SDSS     f4  
    MAG_I_SDSS          f4  
    TRUE_MAG_Z_SDSS_NODUST
                        f4  
    TRUE_MAG_Z_SDSS     f4  
    MAG_Z_SDSS          f4  
    TRUE_FEH            f4  
    TRUE_AGE            f4  
    TRUE_TEFF           f4  
    TRUE_LOGG           f4  
    TRUE_MTIP           f4  
    TRUE_MACT           f4  
    TRUE_SMASS          f4  
    TRUE_POPID          i4  
    IPIX8               i4  
    BRICKNAME           S8  
    BRICKID             i8  
    OBJID               i8  
```

**Comment by apcooper** [comment]
@moustakas: I thought the superfaint mocks were just for photometric contaminants, so I cut out all kinematic information, including the radial velocity. I guess the RV would be used for generating spectra, but is there any need for those to be realistic? If not maybe they could be added on the fly if really needed?

**Comment by moustakas** [comment]
Their primary purpose is to serve as spectral contaminants; sorry for the confusion.  I can add the radial velocity on-the-fly, no problem, but can you recommend a reasonable mean, dispersion, and functional (e.g., Gaussian?) form?

**Comment by apcooper** [comment]
@moustakas: Sorry for the slow response -- your Gaussian is exactly what I was going to suggest.
